open-metadata-implementation/access-services/subject-area/subject-area-api/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/responses/SubjectAreaOMASAPIResponse.java

```diff
@@ -13,201 +20,40 @@
-                /*
-                 Exception responses - note that each exception has the same 4 Exception orientated fields.
-                 Ideally these should be in a superclass. Due to restrictions in the @JsonSubTypes processing it is only possible to have
-                 one level of inheritance at this time.
-                 */
-                @JsonSubTypes.Type(value = ClassificationExceptionResponse.class, name = "ClassificationExceptionResponse"),
-                @JsonSubTypes.Type(value = EntityNotDeletedExceptionResponse.class, name = "EntityNotDeletedExceptionResponse") ,
-                @JsonSubTypes.Type(value = EntityNotPurgedExceptionResponse.class, name = "EntityNotPurgedExceptionResponse") ,
-                @JsonSubTypes.Type(value = RelationshipNotDeletedExceptionResponse.class, name = "RelationshipNotDeletedExceptionResponse") ,
-                @JsonSubTypes.Type(value = RelationshipNotPurgedExceptionResponse.class, name = "RelationshipNotPurgedExceptionResponse") ,
-                @JsonSubTypes.Type(value = FunctionNotSupportedExceptionResponse.class, name = "FunctionNotSupportedExceptionResponse") ,
-                @JsonSubTypes.Type(value = InvalidParameterExceptionResponse.class, name = "InvalidParameterExceptionResponse") ,
-                @JsonSubTypes.Type(value = MetadataServerUncontactableExceptionResponse.class, name = "MetadataServerUncontactableExceptionResponse") ,
-                @JsonSubTypes.Type(value = RelationshipNotDeletedExceptionResponse.class, name = "RelationshipNotDeletedExceptionResponse") ,
-                @JsonSubTypes.Type(value = StatusNotsupportedExceptionResponse.class, name = "StatusNotsupportedExceptionResponse") ,
-                @JsonSubTypes.Type(value = UnrecognizedGUIDExceptionResponse.class, name = "UnrecognizedGUIDExceptionResponse") ,
-                @JsonSubTypes.Type(value = UserNotAuthorizedExceptionResponse.class, name = "UserNotAuthorizedExceptionResponse"),
-                @JsonSubTypes.Type(value = PropertyServerExceptionResponse.class, name = "PropertyServerExceptionResponse"),
-
-                @JsonSubTypes.Type(value = GraphResponse.class, name = "GraphResponse")
-
-        })
-public abstract class SubjectAreaOMASAPIResponse
-{
-    protected int       relatedHTTPCode = 200;
-    protected ResponseCategory responseCategory;
-    protected String    messageId = null;
-    protected String    className = null;
-    protected String    actionDescription = null;
-    protected String    formattedMessage = null;
-
-    /**
-     * Default constructor
-     */
-    public SubjectAreaOMASAPIResponse() {
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class SubjectAreaOMASAPIResponse<R> extends FFDCResponseBase {
+    private List<R> result = new ArrayList<>();
 
+    public void addAllResults(Collection<? extends R> entities) {
+        this.result.addAll(entities);
```

Comment:
When creating the response object, we ourselves must explicitly set the type of returned objects. As will be seen in the picture below - if we set the Category category, then we simply cannot put the Term's objects. And we clearly see the type.
<img width="764" alt="Screenshot 2020-06-24 at 09 04 36" src="https://user-images.githubusercontent.com/35165540/85507763-8cb03b80-b5fb-11ea-84d4-a5436908e782.png">

But also in this case, "? extends R" allows you to specify a generalized object there (it can even be an interface) in order to put the types of objects inherited from it into the response. This shows the following example with Line.

Therefore, no confusion should arise.